src/library_glemu.js

```diff
@@ -175,6 +241,26 @@ var LibraryGLEmulation = {
             GLEmulation.fogEnabled = false;
           }
           return;
+        } else if ((cap >= 0x3000) && (cap < 0x3006)  /* GL_CLIP_PLANE0 to GL_CLIP_PLANE5 */) {
+          var clipPlaneId = cap - 0x3000;
+          if (GLEmulation.clipPlaneEnabled[clipPlaneId] != false) {
+            GLImmediate.currentRenderer = null; // clip plane parameter is part of the FFP shader state, we must re-lookup the renderer to use.
+            GLEmulation.clipPlaneEnabled[clipPlaneId] = false;
+          }
+          return;
+        } else if ((cap >= 0x4000) && (cap < 0x4008)  /* GL_LIGHT0 to GL_LIGHT7 */) {
+          var lightId = cap - 0x4000;
+          if (GLEmulation.lightEnabled[lightId] != false) {
+            GLImmediate.currentRenderer = null; // light parameter is part of the FFP shader state, we must re-lookup the renderer to use.
+            GLEmulation.lightEnabled[lightId] = false;
+          }
+          return;
+        } else if (cap == 0xB50 /* GL_LIGHTING */) {
+          if (GLEmulation.lightingEnabled != false) {
+            GLImmediate.currentRenderer = null; // Fog parameter is part of the FFP shader state, we must re-lookup the renderer to use.
+            GLEmulation.lightingEnabled = false;
+        }
```

Comment:
something looks wrong here and on the line below with the indentation.

Reply:
Embarrassing. Fixed that too.